### ISSUE TITLE
Mongo proxy filter uncommon behavior 

### ISSUE THREAD

**Issue by shiponcs** [envoyproxy/envoy#32289]
In case of MySQL proxy filter,  the [constructor](https://github.com/envoyproxy/envoy/blob/main/contrib/mysql_proxy/filters/network/source/mysql_filter.cc#L22) of the class that inherits the [class Filter](https://github.com/envoyproxy/envoy/blob/main/envoy/network/filter.h#L230) is called once for each connection attempt from downstream and it doesn't get called for subsequent packet exchange on that connection. But in case of Mongo proxy filter, the same [constructor](https://github.com/envoyproxy/envoy/blob/main/contrib/mysql_proxy/filters/network/source/mysql_filter.cc#L22) gets called multiple times which means that corresponding class is initiated multiple times. Then how does this filter handles the information of each connection?
I'm doing some logging stuff in these DB filter for which I am maintaining a singleton class for each connection, The class is holding information about the connection. I am instantiating the class in the above mentioned class constructor. As this constructor is called multiple times for the case of Mongo proxy, it is not working. But my solution worked for MySQL proxy filter.
Please share your knowledge on this. Can I solve this in any other way?

**Comment by ravenblackx** [comment]
@Mythra or @giantcroc as the mongo filter owners?

**Comment by giantcroc** [comment]
> @Mythra or @giantcroc as the mongo filter owners?

Sorry, I haven’t done too much work for mongo proxy and I'm afraid can't help with this issue.

**Comment by shiponcs** [comment]
I have troubleshooted the issue; in fact it's not even an issue—it's the **MongoDB Client** implementation that uses more than one port while communicating with the server and this is how it is intended and designed.

Moreover, **MongoDB client** does connection pooling size of which can be configured using `maxPoolSize` field in the connection string. Yet, even if we set it to `1` , mongo client will still use two ports (meaning two different _TCP connection_ with the server): one to exchange user-put command data with the server and another one for maybe exchanging meta data.

Mongo proxy filter will be instantiated for every TCP connection (different port = different TCP connection) no matter if belongs to the same client-server session or not.
